packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -1477,17 +1477,32 @@ class TextPainter {
     ui.BoxHeightStyle boxHeightStyle = ui.BoxHeightStyle.tight,
     ui.BoxWidthStyle boxWidthStyle = ui.BoxWidthStyle.tight,
   }) {
-    assert(_debugAssertTextLayoutIsValid);
     assert(selection.isValid);
+    return getBoxesForRange(
+      selection.start,
+      selection.end,
+      boxHeightStyle: boxHeightStyle,
+      boxWidthStyle: boxWidthStyle,
+    );
+  }
+
+  /// Returns a list of rects that bound the given range.
```

Comment:
Nit: I wonder if we should move some of the documentation from `getBoxesForSelection` to `getBoxesForRange` and add a mention to each other in the docs.

Reply:
done

Reply:
consider using macro template to share document between these two methods

Reply:
done